### PR TITLE
SSCS-9599: MYA – Split out document and audio/visual evidence uploads

### PR DESCRIPTION
**Pull request by srochani**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java [hunk 1]
```diff
@@ -101,28 +101,49 @@ public CreatedCcdCase createOralCase(String emailAddress) throws IOException {
...
-                Objects.requireNonNull(this.getClass().getClassLoader().getResourceAsStream(fileName)),
-                ContentType.IMAGE_PNG,
-                fileName)
-            .addTextBody("body", description)
-            .addTextBody("idamEmail", "mya-sscs-6920@mailinator.com")
-            .build();
+                .setContentType(ContentType.MULTIPART_FORM_DATA)
+                .addBinaryBody("file",
+                        Objects.requireNonNull(this.getClass().getClassLoader().getResourceAsStream(fileName)),
+                        ContentType.IMAGE_PNG,
```
Comment (divisathyan) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java:109]: Is this type appropriate for audio video file?

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java [hunk 2]
```diff
@@ -101,28 +101,49 @@ public CreatedCcdCase createOralCase(String emailAddress) throws IOException {
...
+    }
+
     public void uploadAppellantStatement(String hearingId, String statement) throws IOException {
         String uri = "/api/continuous-online-hearings/" + hearingId + "/statement";
         String stringEntity = "{\n"
-            + "  \"body\": \"statement\",\n"
-            + "  \"tya\": \"Q9jE2FQuRR\"\n"
-            + "}";
+                + "  \"body\": \"statement\",\n"
+                + "  \"tya\": \"Q9jE2FQuRR\"\n"
```
Comment (divisathyan) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java:145]: What is this code for? Should it be configurable

**Review by divisathyan — COMMENTED** [review]
2 comments

File: src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java [hunk 3]
```diff
@@ -141,28 +148,105 @@ public boolean submitHearingEvidence(String identifier, EvidenceDescription desc
...
+
+                    Document document = evidenceManagementService.upload(convertedFiles, DM_STORE_USER_ID).getEmbedded().getDocuments().get(0);
+
+                    List<E> currentDocuments = documentExtract.getDocuments().apply(caseDetails.getData());
+                    ArrayList<E> newDocuments = (currentDocuments == null) ? new ArrayList<>() : new ArrayList<>(currentDocuments);
+                    newDocuments.add(createNewDocument.apply(document));
+
+                    documentExtract.setDocuments().accept(caseDetails.getData(), newDocuments);
+
+                    ccdService.updateCase(caseDetails.getData(), caseDetails.getId(), eventType.getCcdType(), summary, UPDATED_SSCS, idamService.getIdamTokens());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java:177]: "UPDATED_SSCS" - this description is abit vague...anyway of improving that?

File: src/main/java/uk/gov/hmcts/reform/sscs/controller/EvidenceUploadController.java [hunk 4]
```diff
@@ -33,6 +39,77 @@ public EvidenceUploadController(
...
+    @PutMapping(
+            value = "{identifier}/evidence",
+            consumes = MediaType.MULTIPART_FORM_DATA_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<Evidence> uploadEvidence(
+            @ApiParam(value = "either the online hearing or CCD case id", example = "xxxxx-xxxx-xxxx-xxxx") @PathVariable("identifier") String identifier,
+            @RequestParam("file") MultipartFile file
+    ) {
+        return uploadEvidence(() -> evidenceUploadService.uploadDraftHearingEvidence(identifier, file));
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/controller/EvidenceUploadController.java:61]: Why is this called 'uploadDraftHearingEvidence'? This isn't anything to do with a Hearing right?

Comment (srochani) [reply src/main/java/uk/gov/hmcts/reform/sscs/controller/EvidenceUploadController.java:61]: No, it just uploading the draft document from MYA

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java (hunk 2 above)
Comment (srochani) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java:145]: Its functional test and no change in this functional test

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java (hunk 1 above)
Comment (srochani) [reply src/e2e/java/uk/gov/hmcts/reform/sscs/functional/mya/SscsMyaBackendRequests.java:109]: This method to upload only documents, we have separate method uploadSingleHearingEvidence for audioVideo file upload

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**